src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -152,6 +160,21 @@ public String url() {
     private transient boolean allowKill;
 
     /**
+     * Cumulative list of people who contributed to the build problem.
+     *
+     * <p>
+     * This is a list of {@link User#getId() user ids} who made a change
+     * since the last non-broken build. Can be null (which should be
+     * treated like empty set), because of the compatibility.
+     *
+     * <p>
+     * This field is semi-final --- once set the value will never be modified.
+     *
+     * @since 1.137
+     */
+    private volatile Set<String> culprits;
```

Comment:
Aaaaah. Did not think that through, and that's actually, IMO, a dealbreaker - there's no real value to culprits in Pipeline if you can't do it during the build. So I'll let this stew for the night and see if I've got a better idea tomorrow.